lib/chef/resource/aws_elasticsearch_domain.rb

```diff
@@ -0,0 +1,44 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+module AWS
+  class Elasticsearch
+    class Domain
+    end
+  end
+end
+
+class Chef::Resource::AwsElasticsearchDomain < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type ::AWS::Elasticsearch::Domain
+
+  INSTANCE_TYPES = %w(m3.medium.elasticsearch m3.large.elasticsearch
```

Comment:
We should note in the docs that these are what we support, since presumably Amazon will change the list.